docroot/modules/custom/uiowa_alerts/es6/alerts.es6

```diff
@@ -1,89 +1,129 @@
+          if (this.readyState === 4 && this.status === 200) {
+            handleResponse(this);
+          }
         };
 
-        // Get alerts on page load.
-        updateAlerts(messages);
+        // Set up the parameters for the request and send it.
+        xhttp.open('GET', settings.uiowaAlerts.source, true);
+        xhttp.send();
+      }
+
+      // Handle the response gotten from the alerts endpoint API call.
+      function handleResponse(response) {
+
+        let newAlerts = [];
+
+        // Parse the JSON string into a usable array.
+        const responseJSON = JSON.parse(response.responseText);
+
+        // For each item we find in the JSON data...
+        responseJSON.data.forEach((item, i) => {
+
+          // Create a new ID for it using the UNIX date to avoid collision
+          // and add it to our new alerts
+          const id = `hawk-alert-${item.attributes.date}`;
+          newAlerts.push(id);
+
+          // If it is already in the Drupal messages section, do not continue.
+          if (messages.select(id)) {
+            return;
+          }
+
+          // Get the crafted alert markup
+          const alert = alertMarkup(item);
+
+          // And add it to the Drupal messages section with its unique ID and message type.
+          messages.add(alert, {
+            id: id,
+            type: 'warning'
+          });
+
+          // Look for differences in existing and new alerts and remove any closed alerts.
+          const difference = existingAlerts.filter(existingAlert => !newAlerts.includes(existingAlert));
+          difference.forEach((closed) => {
+            messages.remove(closed);
+          })
+
+          //Then set existing alerts to the new alerts for the next cycle.
+          existingAlerts = newAlerts;
+        })
+      }
+
+      // Gets existing alerts tracked in the messages wrapper.
+      // Returns an array of DOM elements.
+      function getExistingAlerts() {
+        const existing = [];
+        messagesWrapper.querySelectorAll('.messages').forEach( (existingAlert) => {
+          existing.push(existingAlert.getAttribute('data-drupal-message-id'));
+        });
+
+        return existing;
+      }
+
+      // Takes a JSON item and creates the markup for an alert.
+      // Returns a string of HTML.
+      function alertMarkup(responseJSONItem) {
+        const item = responseJSONItem;
+        const date = new Date(item.attributes.date); // parse the ISO 8601 timestamp
+
+        // Create DateTimeFormat instances with the options
+        const monthFormatter = new Intl.DateTimeFormat('en-US', { month: 'long', timeZone: 'America/Chicago' });
+        const timeFormatter = new Intl.DateTimeFormat('en-US', { hour: 'numeric', minute: '2-digit', timeZone: 'America/Chicago', hour12: true });
+
+        // Format the date and time
+        const month = monthFormatter.format(date);
+        const time = timeFormatter.format(date).replace(' AM', 'a.m.').replace(' PM', 'p.m.');
```

Comment:
I think we need to keep the starting space for these, otherwise the time runs together:
![Screenshot 2025-05-28 at 9 43 13 AM](https://github.com/user-attachments/assets/c7caba9b-5822-4c15-b5fb-3ba890919729)
I can add it back in if you'd like.